main.tf

```diff
@@ -217,14 +227,25 @@ resource "aws_cloudwatch_metric_alarm" "cpu_utilization_anomaly_detection" {
 
   metric_query {
     id          = "e1"
-    expression  = "ANOMALY_DETECTION_BAND(m1)"
+    expression  = (local.cpu_utilization == "static_anomaly_detection") ? "ANOMALY_DETECTION_BAND(m2)" : "ANOMALY_DETECTION_BAND(m1)"
     label       = "CPUUtilization (expected)"
     return_data = "true"
   }
 
+  dynamic "metric_query" {
+    for_each = (local.cpu_utilization == "static_anomaly_detection") ? { enabled = true } : {}
+
+    content {
+      id          = "m2"
+      expression  = "IF(m1<${local.cpu_utilization_threshold}, ${local.cpu_utilization_threshold}, m1)"
+      label       = "CPUUtilization (threshold)"
+      return_data = "true"
+    }
+  }
+
   metric_query {
     id          = "m1"
-    return_data = "true"
+    return_data = (local.cpu_utilization == "anomaly_detection") ? "true" : null
```

Comment:
Is it possible, that AWS charges for metric queries, that are part of the alarm definition, even if those are not returning data? Maybe use two different resources for the CloudWatch alarm instead?

Reply:
charges should be the same? we read the same metric. the only difference is that we add another IF() depending on anomaly_detection or static_anomaly_detection

Reply:
You are right, it's per metric not per query.